en/docs/install-and-setup/upgrading-wso2-api-manager/260-to-410/upgrading-from-260-to-410.md

```diff
@@ -0,0 +1,1306 @@
+    
+    -   If you ran WSO2 API-M as a Windows Service when doing the identity component migration , then you need to remove the following parameters in the command line arguments section (CMD_LINE_ARGS) of the api-manager.bat file.
+    
+        ```
+        -Dmigrate -Dcomponent=identity
+        ```
+    
+    -   If you followed step 9 above, and want to use the latest user store, update the <API-M_4.1.0_HOME>/repository/conf/deployment.toml as follows after the identity migration,
+    
+        ```
+        [user_store]
+        type = "database_unique_id"
+        ```
+        
+### Step 4: Migrate the API Manager Components
+
+You have to run the following migration client to update the API Manager artifacts.
+
+1.  Navigate to the [latest release tag (v4.1.0.x)](https://github.com/wso2-extensions/apim-migration-resources/releases) under **Assets** and download the `wso2am-migration-4.1.0.x.zip`. Unzip it to a local directory.
+
+    !!! note
+
+        x of wso2am-migration-4.1.0.x.zip denotes the version number of the most recently-released migration resources.
+
+        The directory where the wso2am-migration-4.1.0.x.zip is unziped will be referred to as `<AM_MIGRATION_CLIENT_HOME>`
+
+2. Copy the `<AM_MIGRATION_CLIENT_HOME>/migration-resources` to the `<API-M_4.1.0_HOME>` directory.
+
+3. Copy the org.wso2.carbon.am.migration-4.1.0.x.jar file in the `<AM_MIGRATION_CLIENT_HOME>/dropins` directory into the `<API-M_4.1.0_HOME>/repository/components/dropins` directory.
+
+4. Prior to API-M migration run the below command to validate the API definitions.
+
+    ``` tab="Linux / Mac OS"
+    sh api-manager.sh -Dmigrate -DmigrateFromVersion=2.6.0 -DmigratedVersion=4.1.0 -DrunPreMigration=apiDefinitionValidation
+    ```
+
+    ``` tab="Windows"
+    api-manager.bat -Dmigrate -DmigrateFromVersion=2.6.0 -DmigratedVersion=4.1.0 -DrunPreMigration=apiDefinitionValidation
+    ```
+
+    Check the server logs and verify if there are any errors logs. If you have encountered any errors in the API definitions, you have to correct them manually on the old version before proceeding to step 5.
+
+5.  Start the API-M server to migrate the API-M components as follows.
+
+    ``` tab="Linux / Mac OS"
+    sh api-manager.sh -Dmigrate -DmigrateFromVersion=2.6.0 -DmigratedVersion=4.1.0
+    ```
+
+    ``` tab="Windows"
+    api-manager.bat -Dmigrate -DmigrateFromVersion=2.6.0 -DmigratedVersion=4.1.0
+    ```
+
+5.  Shutdown the API-M server.
+
+    -   Remove the `org.wso2.carbon.apimgt.migrate.client-4.1.0.x.jar` file, which is in the `<API-M_4.1.0_HOME>/repository/components/dropins` directory.
+
+    -   Remove the `migration-resources` directory, which is in the `<API-M_4.1.0_HOME>` directory.
+
+6. In WSO2 Identity Server 5.11.0, groups include user store roles and roles include internal roles. To enable this role and group separation the following property should be enabled via the deployment.toml file.
```

Comment:
Do we need this step here?

Reply:
Yes.

Reply:
yes, without this step a user with a custom role would not be able to login to portals after migration. The users created by the script has Internal/xxxx roles and that is why we did not get any issues while testing.